h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -103,6 +109,71 @@ public void initMetalearnerParams(Metalearner.Algorithm algo) {
     public Key<Frame>[] _base_model_predictions_keys; 
   }
 
+  public static class StackedEnsembleMRScorer extends MRTask<StackedEnsembleMRScorer> {
+    // IN
+    private Frame scoringFrame;
+    private String seKey;
+    private Job job;
+
+    // OUT
+    private Frame levelOneFrame;
+
+    StackedEnsembleMRScorer(Frame scoringFrame, String seKey, Job j) {
+      this.scoringFrame = scoringFrame;
+      this.seKey = seKey;
+      this.job = j;
+    }
+
+    @Override
+    public void map(Key key) {
+      levelOneFrame = new Frame();
+      Model baseModel = (Model) key.get();
+      Frame basePreds = baseModel.score(
+              scoringFrame,
+              "preds_base_" + seKey + scoringFrame._key,
+              job,
+              false
+      );
+      StackedEnsemble.addModelPredictionsToLevelOneFrame(baseModel, basePreds, levelOneFrame);
+      DKV.remove(basePreds._key); //FIXME: is this necessary? It seems to me that `deleteTempFrameAndItsNonSharedVecs` should take care of it
+      Frame.deleteTempFrameAndItsNonSharedVecs(basePreds, levelOneFrame);
+
+      // FIXME: seems fishy to me and there is probably better solution to this problem
+      // Invoking `FrameUtils.cleanUp` seems to be necessary in multinode setting to prevent leaks.
+      // The leaks are occuring from the keys that were created on the slave[1] nodes - more specifically it appears to be
+      // limited to "adapted missing vectors".
+      /*
+       for (Map.Entry<Key, String>  entry: (java.util.Set < Map.Entry < Key, String>>) baseModel._toDelete.entrySet()) {
+         System.out.println(">>>>> Key ='".concat(entry.getKey().toString()).concat("' value = '").concat(entry.getValue()).concat("'"));
+       }
+      */
+      // prints:
+      /*
+      >>>>> Key ='$04ff11000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'adapted missing vectors' << This is leaked without the cleanup
+      >>>>> Key ='$04ff1c000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'categorically adapted vec'
+      >>>>> Key ='$04ff1d000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'categorically adapted vec'
+      >>>>> Key ='$04ff10000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'adapted missing vectors' << This is leaked without the cleanup
+      */
+      // And leaks both of the "adapted missing vectors" on the slave[1] node.
+      //
+      // [1] By slave node I meant a node that wasn't the one that was directly invoked to run the test
+
+      FrameUtils.cleanUp(baseModel._toDelete);
```

Comment:
@sebhrusen could you look at this. It seems to fix the problem with leaking keys (at least locally) , but I am wondering if it shouldn't be fixed somewhere else.

When I run multi-node tests, `test_SE_scoring_with_missing_response_column` fails with leaked keys. The keys that leak are subset of the ones in `baseModel._toDelete` and AFAIK always the ones with "adapted missing vectors" label and also must be created on slave[1] node. The other keys that are contained in `baseModel._toDelete` but don't leak appear to be recreated on master[1] node (I guess we deterministically generate Vec key names) and deleted afterwards.

 (also please see the comment in the source code for more details)

[1] By master node I mean the one that was invoked to run the test and by slave I mean any other node in the cluster.

Reply:
let's see if you still have leaking keys after using `Key<Frame>` instead of `Frame` above and managing them using DKV + frame update (write_lock, update, unlock)